cmd/byohctl/service/agent.go

```diff
@@ -0,0 +1,602 @@
+				if err := os.Chmod(ByohAgentBinaryPath, 0755); err != nil {
+					utils.LogError("Failed to make binary executable: %v", err)
+				} else {
+					utils.LogSuccess("Made binary executable")
+				}
+			}
+		}
+	}
+
+	// Create the config directory if needed
+	if err := os.MkdirAll(ByohAgentConfDir, 0755); err != nil {
+		utils.LogError("Failed to create config directory: %v", err)
+	}
+
+	// Check service unit file
+	serviceUnitName := ByohAgentServiceName + ".service"
+	unitPathOutput, err := runCommandWithOutput("systemctl", "show", "-p", "FragmentPath", serviceUnitName)
+	if err != nil {
+		utils.LogError("Failed to find service unit path: %v", err)
+	} else {
+		utils.LogInfo("Service unit file: %s", string(unitPathOutput))
+
+		// Try to get unit file content
+		parts := strings.Split(string(unitPathOutput), "=")
+		if len(parts) > 1 {
+			unitPath := strings.TrimSpace(parts[1])
+			if unitFileContent, err := os.ReadFile(unitPath); err == nil {
+				utils.LogInfo("Service unit file content:\n%s", string(unitFileContent))
+			}
+		}
+	}
+
+	// Check systemd service status
+	statusOutput, err := runCommandWithOutput("systemctl", "status", serviceUnitName)
+	if err != nil {
+		utils.LogWarn("Service %s is not running: %v", serviceUnitName, err)
+		utils.LogInfo("Service status output: %s", statusOutput)
+
+		// Give suggestion on how to start
+		utils.LogInfo("Suggestion: Try starting the service manually with:")
+		utils.LogInfo("    sudo systemctl start %s", serviceUnitName)
+		utils.LogInfo("    sudo systemctl enable %s", serviceUnitName)
+	} else {
+		utils.LogSuccess("Service %s is active", serviceUnitName)
+	}
+
+	// Check logs regardless of service status
+	journalOutput, err := runCommandWithOutput("journalctl", "-u", serviceUnitName, "--no-pager", "-n", "20")
+	if err != nil {
+		utils.LogWarn("Unable to fetch service logs: %v", err)
+	} else {
+		utils.LogInfo("Recent logs:")
+		utils.LogInfo("%s", string(journalOutput))
+	}
+
+	return nil
+}
+
+// downloadDebianPackage downloads the debian package from the container registry
+func downloadDebianPackage(outputDir string) (string, error) {
+	utils.LogInfo("Preparing BYOH agent Debian package from %s", ByohAgentDebPackageURL)
+
+	// Check if imgpkg is available (this is now handled by ensureRequiredPackages)
+	imgpkgPath, err := exec.LookPath("imgpkg")
+	if err != nil {
+		return "", fmt.Errorf("imgpkg not found in PATH: %v", err)
+	}
+
+	// Use a buffer to capture the command output
+	var outputBuffer bytes.Buffer
+	pullCmd := exec.Command(imgpkgPath, "pull", "-i", ByohAgentDebPackageURL, "-o", outputDir)
+	pullCmd.Stdout = &outputBuffer
+	pullCmd.Stderr = &outputBuffer
+
+	if err := pullCmd.Run(); err != nil {
+		output := outputBuffer.String()
+
+		if strings.Contains(output, "UNAUTHORIZED") {
+			utils.LogWarn("Authentication error. If this is a private repository, you may need to configure Docker credentials.")
```

Comment:
We know we aren't pulling from a private registry. Why do we need this check?

Reply:
Ohh yes, this is a public repo....We don't need this check. Addressed it in the recent commit